### DISCUSSION TITLE
Rebuilding the model all the time causes memory issues

### DISCUSSION THREAD

**Discussion by 32bit** [Automattic/mongoose#11280] [Q&A] [answered]
I have a situation where my collection name is embedded with the Model creation. This is because we have a versioning process to change collection names at runtime, the model remains the same. 

const mymodel = mongoose.model('TestModel', TestSchema, collectionName);

In production we noted that there was a large footprint to this- a very large number of models are in memory. 
![image](https://user-images.githubusercontent.com/2686464/151252832-a6aff149-4d62-4866-adfd-e286b4989070.png)

Is there a way to compile this once and reset the collectionname when it changes instead of doing that all the time?

**Comment by vkarpov15** [comment, marked as answer]
Not currently. You should be able to reduce the memory usage by deleting the model before creating a new one.

```javascript
mongoose.deleteModel('TestModel');
const mymodel = mongoose.model('TestModel', TestSchema, collectionName);
```
